### PR TITLE
[SeiDB] Fix PebbleDB Iterator Issue

### PR DESCRIPTION
**Pull request by Kbhat1**
## Describe your changes and provide context
- Fix for PebbleDB Iterator issue
- Previously, the mvcc iterator would loop indefinitely when it reached a key that was greater than the iterator version specified and go back to a previous key
- Catch the scenario where iterator moves back to previous key and move it forward to avoid a loop

## Testing performed to validate your change
- Tested and repro'd error on node 

Before fix: 
image

After fix:
image

### CODE REVIEW

File: ss/test/storage_test_suite.go [hunk 1]
```diff
@@ -369,6 +369,34 @@ func (s *StorageTestSuite) TestDatabaseIteratorMultiVersion() {
 	s.Require().NoError(itr.Error())
 }
 
+// Tests bug where iterator loops continuously
+func (s *StorageTestSuite) TestDatabaseIteratorLooping() {
```
Comment (yzang2019) [inline ss/test/storage_test_suite.go:373]: We should add a test for reverse iteration as well

File: ss/pebbledb/iterator.go [hunk 2]
```diff
@@ -116,15 +116,15 @@ func (itr *iterator) Value() []byte {
...
-		currKey, _, ok := SplitMVCCKey(itr.source.Key())
-		if !ok {
-			// XXX: This should not happen as that would indicate we have a malformed
-			// MVCC key.
-			panic(fmt.Sprintf("invalid PebbleDB MVCC key: %s", itr.source.Key()))
-		}
-
 		// Since PebbleDB has no PrevPrefix API, we must manually seek to the next
 		// key that is lexicographically less than the current key.
 		next = itr.source.SeekLT(MVCCEncode(currKey, 0))
```
Comment (yzang2019) [inline ss/pebbledb/iterator.go:130]: would reverse iteration also be a problem?

File: ss/pebbledb/iterator.go [hunk 3]
```diff
@@ -154,6 +154,27 @@ func (itr *iterator) Next() {
...
+			itr.valid = false
+			return
+		}
+
+		// There exists cases where the SeekLT() call moved us back to the same key
+		// we started at, so we must move to next key, i.e. two keys forward.
+		if bytes.Equal(tmpKey, currKey) {
+			if itr.source.SeekGE(MVCCEncode(nextKey, 0)) {
+				if itr.source.NextPrefix() {
+					itr.Next()
```
Comment (yzang2019) [inline ss/pebbledb/iterator.go:170]: Why do we move to next again after seeking GE, will that result in skipping a key that we should iterate?

File: ss/test/storage_test_suite.go (hunk 1 above)
Comment (Kbhat1) [reply ss/test/storage_test_suite.go:373]: Yea there are a lot more unit tests that are going to be pushed

**Review by yzang2019 — APPROVED** [review]

File: ss/pebbledb/iterator.go (hunk 2 above)
Comment (Kbhat1) [reply ss/pebbledb/iterator.go:130]: Reverse wouldn't be a problem (discussed this offline)

**Pull request merged**